quinn/dataframe_validator.py

```diff
@@ -36,14 +36,14 @@ def validate_presence_of_columns(df: DataFrame, required_col_names: list[str]) -
     error_message = f"The {missing_col_names} columns are not included in the DataFrame with the following columns {all_col_names}"
     if missing_col_names:
         raise DataFrameMissingColumnError(error_message)
-    
+
 def validate_schema(
-    required_schema: StructType, 
-    ignore_nullable: bool = False, 
-    _df: DataFrame = None
-) -> function:
+    required_schema: StructType,
+    ignore_nullable: bool = False,
+    _df: DataFrame = None,
```

Comment:
Why are we using private variables (I mean `_df`) naming convention for a public API (I mean function arguments)?

Reply:
Duh! Sorry, I meant to change this and completely forgot. Let me fix this. 

Meanwhile, can we not have `ruff-format` as one of the pre-commit hooks? First - It's experimental and called out so; second - it seems to be reformatting a whole lot of files which are not part of this PR when I run it on local. 
<img width="1052" alt="image" src="https://github.com/user-attachments/assets/e5daa431-f277-495c-93c1-7c254676be3a">

Reply:
@SemyonSinchenko Renamed `_df` to `df_to_be_validated`

Reply:
@fpgmaas May you take a look, please?

Reply:
@kunaljubce In the [CI/CD pipeline](https://github.com/MrPowers/quinn/actions/runs/9953381549/job/27496705676#step:4:58) I see that only 1 file is improperly formatted. I see your `.pre-commit-config.yaml` contains unstaged changes, my guess that is causing your issue. Why is that changes, and what does the `pre-commit-config.yaml` look like?

Reply:
@fpgmaas The unstaged changes are because I added `    - id: ruff-format` to my `pre-commit-config.yaml`. Here's a screen recording of my changes and pre-commit succeeding before `ruff-format` v/s pre-commit failing and fixing 9 files after `ruff-format` - https://ufile.io/792yfg0c

Reply:
I don't think you should edit that file manually, did you try pulling or rebasing on top of `[MrPowers:planning-1.0-release](https://github.com/MrPowers/quinn/tree/planning-1.0-release)`? 

If I look at your branch, you are also still using an outdated version of ruff, see [here](https://github.com/kunaljubce/quinn/blob/kunaljubce/add-decorators-for-functions/.pre-commit-config.yaml)

Reply:
@fpgmaas That was my mistake. Out of instinct, I rebased this branch with `main` and now retrying to rebase it with `planning-1.0-release` has corrupted this PR. 😭 

Closing this and reopening a fresh PR once more.

Reply:
@kunaljubce Ah, that explains! Understandable mistake :) Good you were able to figure it out.